### PR TITLE
Javier fetch override callbacks

### PR DESCRIPTION
**Pull request by JavierFerrerGomez**
## Description, Motivation and Context

I expose 3 functions to JavaScript that can be use as callbacks.

* Link to relevant: #54

## How Has This Been Tested?
Relevant tests in Test_Fetch.cpp had been updated. So you can do:
```
cd niLang
. hat
ham Run_Test_niCURL
```

To test the web version you can follow the same as in pr #54

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that cause existing functionality to change)
* [ ] Documentation (code docs, comments, or changes to the doc systems)
* [ ] Testing/Build (test coverage or the test/build subsystems themselves)
* [ ] Packaging (adds examples or modifies a release package)

### CODE REVIEW

File: sources/niCURL/src/_build.ham [hunk 1]
```diff
@@ -12,6 +12,12 @@ if ! ( $(USE_SYSTEM_LIBCURL) = 1 ) {
   SubDirHdrs $(SUBDIR) libCURL include ;
 }
```
Comment (JavierFerrerGomez) [inline sources/niCURL/src/_build.ham:14]: I really dont know how to do this...I tried a lot of different ways. I just want to se extra linking flags ONLY FOR niCURL.
I tried LINKFLAGS on niCURL, LINKFLAGS on $(PKGTARGET), TargetLinkFlags $(<) : and also creating _rules.ham in the niCURL folder and overrite the flags there.

As I have it this PR it works if I run ham all inside the niCURL directory, the problem I have is that when I run ham all in the root of niLang, other projects like niUI are trying to link _FetchOverride_Error also.

File: sources/niCURL/src/_build.ham [hunk 2]
```diff
@@ -12,6 +12,12 @@ if ! ( $(USE_SYSTEM_LIBCURL) = 1 ) {
...
 
+if ( $(EMSCRIPTEN) = 1 ) {
+  LINKFLAGS ?= $(LINKFLAGS) ;
+  LINKFLAGS += -s EXPORTED_FUNCTIONS=_main,_malloc,_free,_FetchOverride_Success,_FetchOverride_Error,_FetchOverride_Progress
+    ;
+}
+
```
Comment (JavierFerrerGomez) [inline sources/niCURL/src/_build.ham:20]: I really dont know how to do this...I tried a lot of different ways. I just want to se extra linking flags ONLY FOR niCURL.
I tried LINKFLAGS on niCURL, LINKFLAGS on 
(<) : and also creating _rules.ham in the niCURL folder and overrite the flags there.

As I have it this PR it works if I run ham all inside the niCURL directory, the problem I have is that when I run ham all in the root of niLang, other projects like niUI are trying to link _FetchOverride_Error also.

Comment (prenaux) [inline sources/niCURL/src/_build.ham:20]: This is how it's done in niUI:
```
if $(ANDROID) = 1 {
  LINKLIBS on $(PKGTARGET) ?= $(LINKLIBS) ;
  LINKLIBS on $(PKGTARGET) += -lGLESv2 ;
}
```

`s/LINKLIBS/LINKFLAGS/g` should work...

Comment (prenaux) [reply sources/niCURL/src/_build.ham:20]: I think what you want is to add this in _rules_sdks.ham at https://github.com/prenaux/niLang/blob/3a64369f19c604b8193c745fe3159680fbfb81db/sources/_rules_sdks.ham#L226

**Comment by prenaux** [comment]
Deprecated, see #70

**Pull request closed without merging**